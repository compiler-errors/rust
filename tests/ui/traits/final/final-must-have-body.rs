#![feature(final_associated_functions)]

trait Foo {
    final fn method();
    //~^ ERROR `final` is only allowed on associated functions if they have a body
}

fn main() {}
