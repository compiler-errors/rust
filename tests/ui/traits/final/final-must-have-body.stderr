error: `final` is only allowed on associated functions if they have a body
  --> $DIR/final-must-have-body.rs:4:5
   |
LL |     final fn method();
   |     -----^^^^^^^^^^^^^
   |     |
   |     `final` because of this

error: aborting due to 1 previous error

